cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -240,6 +286,47 @@ ReconciliationState createReconciliationState(Reconciliation reconciliation, Kaf
         return new ReconciliationState(reconciliation, kafkaAssembly);
     }
 
+    Future<Void> updateStatus(Reconciliation reconciliation, String namespace, String name, KafkaStatus desiredStatus) {
```

Comment:
When working on it before, I though that it does not fit so well into the `ReconciliationState` because of what it does. But I'm not sure now what I saw. I will move it.